src/Illuminate/Broadcasting/Broadcasters/AblyBroadcaster.php

```diff
@@ -139,62 +193,114 @@ public function broadcast(array $channels, $event, array $payload = [])
+    {
+        $header = [
+            'typ' => 'JWT',
+            'alg' => 'HS256',
+            'kid' => $this->getPublicToken(),
+        ];
+        // Set capabilities for public channel as per https://ably.com/docs/core-features/authentication#capability-operations
+        $channelClaims = $this->defaultChannelClaims;
+        $serverTimeFn = function () {
+            return self::getServerTime();
+        };
+        if ($token && Utils::isJwtValid($token, $serverTimeFn, $this->getPrivateToken())) {
+            $payload = Utils::parseJwt($token)['payload'];
+            $iat = $payload['iat'];
+            $exp = $payload['exp'];
+            $channelClaims = json_decode($payload['x-ably-capability'], true);
+        } else {
+            $iat = $serverTimeFn();
+            $exp = $iat + $this->tokenExpiry;
+        }
+        if ($channelName) {
+            $channelClaims[$channelName] = $channelCapability;
+        }
+        $claims = [
+            'iat' => $iat,
+            'exp' => $exp,
+            'x-ably-clientId' => $clientId ? strval($clientId) : null,
+            'x-ably-capability' => json_encode($channelClaims),
+        ];
+
+        return Utils::generateJwt($header, $claims, $this->getPrivateToken());
+    }
+
+    /**
+     * Remove prefix from channel name.
      *
-     * @param  string  $event
-     * @param  array  $payload
-     * @return \Ably\Models\Message
+     * @param $channel string
+     * @return string
      */
-    protected function buildAblyMessage($event, array $payload = [])
+    public function normalizeChannelName($channel)
     {
-        return tap(new AblyMessage, function ($message) use ($event, $payload) {
-            $message->name = $event;
-            $message->data = $payload;
-            $message->connectionKey = data_get($payload, 'socket');
-        });
+        if ($channel) {
+            if ($this->isPrivateChannel($channel)) {
+                return Str::replaceFirst('private:', '', $channel);
+            }
+            if ($this->isPresenceChannel($channel)) {
+                return Str::replaceFirst('presence:', '', $channel);
+            }
+
+            return Str::replaceFirst('public:', '', $channel);
+        }
+
+        return $channel;
     }
 
     /**
-     * Return true if the channel is protected by authentication.
+     * Checks if channel is a private channel.
      *
-     * @param  string  $channel
+     * @param $channel string
      * @return bool
      */
-    public function isGuardedChannel($channel)
+    public function isPrivateChannel($channel)
     {
-        return Str::startsWith($channel, ['private-', 'presence-']);
+        return Str::startsWith($channel, 'private:');
     }
 
     /**
-     * Remove prefix from channel name.
+     * Checks if channel is a presence channel.
      *
-     * @param  string  $channel
-     * @return string
+     * @param $channel string
```

Comment:
```suggestion
     * @param  string  $channel
```
First variable type and then the variable name

Reply:
Fixed!